### PR TITLE
Add tests for SomeSerialisationConstraint 

### PR DESCRIPTION
**Pull request by phadej**
- Include "not-so-trivial" instance for Word64

This will conflict with #7, but I'll resolve the conflicts whichever is merged first.

### CODE REVIEW

**Review by jorisdral — APPROVED** [review]

File: src/Database/LSMTree/Model/Normal.hs [hunk 1]
```diff
@@ -51,10 +53,28 @@ class SomeSerialisationConstraint a where
     -- Note: cannot fail.
     deserialise :: BS.ByteString -> a
 
+-- Note: we might add extra requirements,
+-- as this instance may produce arbitrary large serialised values.
+-- Do we need to limit maximum serialised size?
+-- Or even require it to be constant (and known in advance?)
```
Comment (dcoutts) [inline src/Database/LSMTree/Model/Normal.hs:59]: Yes, this is a good point, but we expect the key size constraints to be part of the table config, not global for the key type. In particular, we expect to have table keys be a fixed size, but with the fixed size different for different tables, and values to have some maximum size (though implementation defined not per-table).

File: src/Database/LSMTree/Model/Normal.hs [hunk 2]
```diff
@@ -51,10 +53,28 @@ class SomeSerialisationConstraint a where
...
+        b8 = fromIntegral $        w    .&. 0xff
+        b7 = fromIntegral $ shiftR w  8 .&. 0xff
+        b6 = fromIntegral $ shiftR w 16 .&. 0xff
+        b5 = fromIntegral $ shiftR w 24 .&. 0xff
+        b4 = fromIntegral $ shiftR w 32 .&. 0xff
+        b3 = fromIntegral $ shiftR w 40 .&. 0xff
+        b2 = fromIntegral $ shiftR w 48 .&. 0xff
+        b1 = fromIntegral $ shiftR w 56 .&. 0xff
+
+    deserialise = BS.foldl' (\acc d -> acc * 0x100 + fromIntegral d) 0
```
Comment (dcoutts) [inline src/Database/LSMTree/Model/Normal.hs:76]: We should add an optimisation TODO here. We should be able to use some primops to load and endian-swapping primops.

One thing we'll decide later when we do the file format details is if we guarantee keys to have aligned addresses or not.

**Review by dcoutts — APPROVED** [review]
Yes, nice.

File: src/Database/LSMTree/Model/Normal.hs (hunk 1 above)
Comment (phadej) [reply src/Database/LSMTree/Model/Normal.hs:59]: Isn't that what's kind of what `newtype`s are for?

File: src/Database/LSMTree/Model/Normal.hs (hunk 2 above)
Comment (phadej) [reply src/Database/LSMTree/Model/Normal.hs:76]: Note: this is *model* `SomeSerializationConstraint`.

First you (and @jorisdral) have to agree/decide that we want or don't want to have only one serialization constraint.

**Pull request merged**